app/Http/Controllers/Api/PostController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Models\Post;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+use App\Http\Requests\PostRequest;
+use App\Http\Controllers\Controller;
+use App\Http\Resources\PostResource;
+use Illuminate\Support\Facades\Validator;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index(Request $request): JsonResponse
+    {
+        $pageSize = $request->input('pageSize', 10);
```

Comment:
Add a validator to check the input then 
The user should be able to select the specific page he/she desires 

$page = $validated['page'] // by default this is 0
$size = $validated['size'] // by default this can be 10 or 20 Depending on you!

$data = Model::query()->offSet($page * $size)->take($size)
remove paginate in the api we wont use paginate